Bluetooth: Controller: Fix read ISO Tx Sync for Broadcast ISO

cvinayak opened the pull request:
Fix missing read ISO Tx Sync for Broadcast ISO feature.

Correction to lll_adv_iso->iso_interval value stored.

Fixes #56203.

cvinayak committed:
Bluetooth: Controller: Fix read ISO Tx Sync for Broadcast ISO

Fix missing read ISO Tx Sync for Broadcast ISO feature.

Co-authored-by: Nirosharn Amarasinghe <niag@demant.com>
Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
@niag-Oticon need your review here, I reused the way ISO data is handled in CIS for BIS.

niag-Oticon commented on subsys/bluetooth/controller/ll_sw/ull_adv_iso.c:
```diff
@@ -1291,6 +1292,14 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
 	LL_ASSERT(!ret);
 
+	/* Calculate the BIG reference point of current BIG event */
+	remainder_us = remainder;
```
Is remainder in microseconds?

cvinayak replied:
Remainder is in HAL specific unit (+/-30.517 us stored in picoseconds to fit 32-bits). `hal_ticker_remove_jitter()`  is used to get a positive remainder in microseconds by adjusting the `ticks_to_expire` accordingly.

cvinayak commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -5817,33 +5817,12 @@ int hci_iso_handle(struct net_buf *buf, struct net_buf **evt)
-		sdu_frag_tx.target_event = pkt_seq_num;
-
-		stream->pkt_seq_num++;
+		sdu_frag_tx.target_event = lll_iso->payload_count / lll_iso->bn;
```
`lll_iso->payload_count`  is the value for the next BIG event, hence, the target_event is correct.

niag-Oticon replied:
Is that true even if the SDU Interval is less that the ISO interval? i.e. there are multiple SDU per ISO interval?

I guess I have two concerns which I am not sure are entirely valid:
-- Is there a possibility that this floor division yields an event that is already in progress?
-- In CIS we check ull_ref_get(&cig->ull) to decide if we are past the subevents for the current event. This is done to avoid a situation where the SDU is received too close to the start of the next event such that the fragmentation and propagation could result in payload being too late for transmission. Is this a concern for BIS as well?

cvinayak replied:
Thanks for explaining, i have added your comment in the code.

cvinayak commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -5817,33 +5817,12 @@ int hci_iso_handle(struct net_buf *buf, struct net_buf **evt)
+		sdu_frag_tx.grp_ref_point =
+			isoal_get_wrapped_time_us(adv_iso->big_ref_point,
+						  (lll_iso->iso_interval *
+						   ISO_INT_UNIT_US));
```
`big_ref_point` corresponds to previously expired BIG event, hence the group reference point is correct.

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_adv_types.h:
```diff
@@ -101,6 +101,8 @@ struct ll_adv_iso_set {
 	struct ull_hdr        ull;
 	struct lll_adv_iso    lll;
 
+	uint32_t big_ref_point;
```
Would it make sense to add the unit of this? If not in the name, then perhaps at least in a comment

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_iso.c:
```diff
@@ -176,9 +176,18 @@ uint8_t ll_read_iso_tx_sync(uint16_t handle, uint16_t *seq,
 
 	} else if (IS_ADV_ISO_HANDLE(handle)) {
-		/* FIXME: Do something similar to connected */
+		struct lll_adv_iso_stream *adv_stream;
```
```suggestion
		const struct lll_adv_iso_stream *adv_stream;
```

Thalley reviewed (changes requested):
Minor comments, otherwise LGTM

cvinayak committed:
Bluetooth: Controller: Fix Broadcast ISO interval unit

Fix incorrectly stored Broadcast ISO interval in
microseconds instead of storing in 1.25 ms units.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
tests: bsim: Bluetooth: Fix BIS test ISO Tx Buffers in Controller

Fix BIS test for ISO Tx Buffers in Controller to be 2 for
ISO TX MTU size of 502 bytes.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>
